### PR TITLE
INTMDB-183: Migrate to TF SDK 2

### PR DESCRIPTION
**Pull request by coderGo93**
## Description
- Migrated to SDK Version 2
- Because of bad practice and is not supported for sdk v2, using TypeMap as object will show error with `make test`, so the fix was adding another attribute and deprecate the old attribute, and for the new attributes was changed to TypeList with MaxItem 1, resources/datasources affected are `alert_configuration`, `cloud_provider_snapshot_restore_job`, `mongodbatlas_encryption_at_rest`

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the Terraform contribution guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: website/docs/guides/1.0.0-upgrade-guide.html.markdown
Comment (themantissa) [inline website/docs/guides/1.0.0-upgrade-guide.html.markdown]: Replace "Actual usage" with "New usage" for all please.

File: website/docs/guides/1.0.0-upgrade-guide.html.markdown
Comment (themantissa) [inline website/docs/guides/1.0.0-upgrade-guide.html.markdown]: Just a rewrite to make a bit more clear, please replace this with:
Due to the migration to Terraform SDK v2 various resources and datasource(s) have been impacted and will require a change within your configuration.  These resources and datasources used the now unsupported TypeMap object.  In SDK v2 these must be TypeSet or TypeList with MaxItems set to 1.    The following is the list of resources and datasources impacted by the migration.

**Review by themantissa — CHANGES_REQUESTED** [review]
Just a few docs edits, otherwise looks good.   I'll create a ticket to update search and backup after they are merged.  Just a note for the future - let's break these huge PRs up. For example this could have been a PR for the examples, a PR for docs, and a PR for the work (which we could maybe break up?)  Will just help with review speed and accuracy.

**Review by themantissa — APPROVED** [review]
Thank you for the fixes!  LGTM!

**Comment by themantissa** [comment]
Adding @threebee because @andreaangiolillo is OOO.   @threebee this is ready for review.

File: mongodbatlas/resource_mongodbatlas_alert_configuration.go
Comment (thetonymaster) [inline mongodbatlas/resource_mongodbatlas_alert_configuration.go:576]: This shoundn't be > 0?

Comment (thetonymaster) [inline mongodbatlas/resource_mongodbatlas_alert_configuration.go]: This shouldn't be `> 0`?

File: mongodbatlas/resource_mongodbatlas_alert_configuration.go
Comment (thetonymaster) [inline mongodbatlas/resource_mongodbatlas_alert_configuration.go:593]: If `theshold` is going to be deprecated check first for `threshold_config`

**Review by thetonymaster — APPROVED** [review]
I can't find any major issues besides a couple of nitpicks

File: mongodbatlas/data_source_mongodbatlas_cloud_provider_access_setup.go
Comment (threebee) [inline mongodbatlas/data_source_mongodbatlas_cloud_provider_access_setup.go:39]: [Question] For my own understanding and learning. Why use `TypeList with maxlen 1` instead of TypeStr?

**Review by threebee — APPROVED** [review]
LGTM, if test pass.

This PR was too big to review for me. I'm afraid that I might have missed something.

File: mongodbatlas/data_source_mongodbatlas_cloud_provider_access_setup.go
Comment (coderGo93) [reply mongodbatlas/data_source_mongodbatlas_cloud_provider_access_setup.go:39]: Because TypeList with MaxItems 1 ideally it's a object from the API, and TypeSet is an unordered list

Comment (threebee) [reply mongodbatlas/data_source_mongodbatlas_cloud_provider_access_setup.go:39]: ah, I miss typed it. I meant why not use `TypeString`?

Comment (coderGo93) [reply mongodbatlas/data_source_mongodbatlas_cloud_provider_access_setup.go:39]: :O, I misread it then haha, well, because `aws_config` is a object with two fields

**Pull request merged**
